test/channels/retro_channel_test.exs

```diff
@@ -172,4 +172,29 @@ defmodule RemoteRetro.RetroChannelTest do
       assert_broadcast("idea_highlighted", %{"id" => 1, "isHighlighted" => false})
     end
   end
+
+  describe "pushing a `submit_vote` event to the socket" do
+    setup [:persist_user_for_retro, :persist_idea_for_retro, :join_the_retro_channel]
+
+    @tag user: @mock_user
+    @tag idea: %Idea{category: "sad", body: "JavaScript"}
+    test "results in the broadcast of the voted on idea to all connected clients", %{socket: socket, idea: idea} do
+      idea_id = idea.id
+      push(socket, "submit_vote", %{id: idea_id})
+
+      assert_broadcast("vote_submitted", %{body: "JavaScript", id: ^idea_id, vote_count: 1})
+    end
+
+
+    @tag user: @mock_user
+    @tag idea: %Idea{category: "sad", body: "doggone keeper"}
+    test "results in the idea being updated in the database", %{socket: socket, idea: idea} do
+      idea_id = idea.id
+      push(socket, "submit_vote", %{id: idea_id})
+
+      :timer.sleep(50)
```

Comment:
not ideal, but i recall having to do something similar, due to the db insertion being fired off in another process by the channel